tests/components/element-editor/basic-details/basic-details.spec.tsx

```diff
@@ -18,75 +18,139 @@ describe("BasicDetails", () => {
+        "conditional rendering of element-specific basic details component",
+        ({ elementType, testId }) => {
+            it(`should render the ${elementType} basic details component`, async () => {
+                renderComponent({
+                    builderContext: {
+                        focusedElement: {
+                            element: MOCK_FOCUSED_ELEMENT(elementType),
+                            isDirty: false,
+                        },
+                        selectedElementType: elementType,
+                    },
+                });
+
+                const basicDetails = await screen.findByTestId(testId);
+                expect(basicDetails).toBeInTheDocument();
             });
-            const labelField = await getLabelField();
-            expect(labelField).toBeInTheDocument();
-        });
+        }
+    );
 
-        it("should render required error message if element has required error message property", async () => {
+    describe("resetting of form fields when element type changes", () => {
+        it("should replace element name, mandatory field's error message, and ID with values that correspond to the new element type", async () => {
             renderComponent({
                 builderContext: {
-                    focusedElement: MOCK_FOCUSED_ELEMENT,
+                    focusedElement: {
+                        element: MOCK_FOCUSED_ELEMENT(EElementType.EMAIL),
+                        isDirty: false,
+                    },
                     selectedElementType: EElementType.EMAIL,
                 },
+                formContext: {
+                    defaultValues: MOCK_FOCUSED_ELEMENT(EElementType.EMAIL),
+                },
             });
-            fireEvent.click(await screen.findByLabelText("Yes"));
 
-            const requiredErrorMessageField =
-                await screen.findByLabelText("Error message");
-            expect(requiredErrorMessageField).toBeInTheDocument();
-        });
+            // Check initial state
 
-        it("should render the description if element has the description property", async () => {
-            renderComponent({
-                builderContext: {
-                    focusedElement: MOCK_FOCUSED_ELEMENT,
-                    selectedElementType: EElementType.EMAIL,
-                },
-            });
+            const textBasicDetails = await screen.findByTestId(
+                "email-basic-details"
+            );
+            expect(textBasicDetails).toBeInTheDocument();
+
+            let elementName = screen.getByTestId("label-field");
+            let errorMessageField = screen.getByTestId(
+                "required-error-message-field"
+            );
+            let elementId = screen.getByTestId("id-field");
+
+            expect(elementName).toHaveValue("Email address");
+            expect(errorMessageField).toHaveValue("This is a required field.");
+            expect(elementId).toHaveValue("email-field");
```

Comment:
you can move these assertions into a separate test to make sure the initial state is assigned properly

in the subsequent tests, you can skip these assertions since they have been covered